test/integration/plugin/tests/test-plugin-launch.hpp

```diff
@@ -0,0 +1,85 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-22, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// Header file containing basic integration tests for plugins with launch.
+///
+
+#ifndef __TEST_PLUGIN_LAUNCH_HPP__
+#define __TEST_PLUGIN_LAUNCH_HPP__
+
+#include "test-plugin.hpp"
+
+
+// Check that the plugin is called with the right Platform.
+// Check that the plugin is called the correct number of times,
+// once before and after each launch capture for the capture counter,
+// once before and after each launch invocation for the launch counter.
+
+// test with basic launch
+template <typename LaunchPolicy,
+          typename WORKING_RES,
+          RAJA::Platform PLATFORM>
+void PluginLaunchTestImpl()
+{
+  SetupPluginVars spv(WORKING_RES::get_default());
+
+  CounterData* data = plugin_test_resource->allocate<CounterData>(10);
+
+  for (int i = 0; i < 10; i++) {
+
+    PluginTestCallable p_callable{data};
+
+    RAJA::launch<LaunchPolicy>
+      (RAJA::LaunchParams(RAJA::Teams(1), RAJA::Threads(1)),
+      [=] RAJA_HOST_DEVICE(RAJA::LaunchContext ctx)
+      {
+        p_callable(i);
+      });
```

Comment:
I recommend putting this section in its own scope so that p_callable is destroyed before the asserts. It doesn't affect anything now but it would be a more consistent with the other plugin tests.